processing/src/main/java/org/apache/druid/guice/StartupInjectorBuilder.java

```diff
@@ -47,18 +48,15 @@ public StartupInjectorBuilder()
         new JacksonModule(),
         new ConfigModule(),
         new NullHandlingModule(),
+        new StorageConnectorModule(),
```

Comment:
Do we want this here? Does every test and client need this module? Does it need to be in the startup injector?

Or, would this be better in the `forServer()` method of the `CoreInjectorBuilder`? If any connector needs to have anything injected, then it has to be part of the "second tier" of modules.

The goal with these builders is to let test and clients use Guice, without being forced to also accept server-only components. So, for each new component, ask if tests (both UTs and ITs) need the component? Or, only the server?